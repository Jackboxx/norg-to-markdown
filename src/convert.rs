use crate::{parse::parse, post_process::process};

pub fn convert(text: &str) -> Result<String, regex::Error> {
    let parsed: String = text
        .lines()
        .map(|l| parse(l).unwrap_or_else(|_| "???".to_owned()))
        .collect::<Vec<String>>()
        .join("\n");

    process(&parsed)
}
