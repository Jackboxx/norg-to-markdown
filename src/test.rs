#[cfg(test)]
mod tests {
    use crate::convert::convert;

    #[test]
    fn test_convert() {
        let text = "
* heading
some text
   some text that shouldn't be indented

*fake heading
   - bullet point 1
   - bullet point 2
   - bullet point 3
   - *bold bullet point*
      - sub bullet point 1
         - sub sub bullet point 1
         - sub sub bullet point 2
            - sub sub sub bullet point 1
   -fake bullet point

   ~ order list entry 1
   ~ order list entry 2

some text in between with bold italic */highlights/*.

   ~ level 1 order list entry 1
   ~ level 1 order list entry 2
      ~ level 2 order list entry 1

   ~ order list entry 1
   - unorder list entry 1 with a {:link:}
   ~ order list entry 1

      ~ level 2 order list entry 1
      ~ level 2 order list entry 2
         ~ level 3 order list entry 1

** To Do 
normal text.
{:a link to a file:}
/italic/ and *bold* text!

** Checklist
   - () unchecked
   - (x) checked
   - (=) pending
   - (?) more info needed
   - (_) dropped
   - ( ) not done

@code
const var = 'this is some code';
console.log(var);
@end

@code rust
let var = \"this is some better code\";
println!(\"{var}\");
@end
";

        let res = convert(text).unwrap();
        insta::assert_snapshot!(res);
    }
}
