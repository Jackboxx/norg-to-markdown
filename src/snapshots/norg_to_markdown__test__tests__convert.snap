---
source: src/test.rs
expression: res
---
# heading
some text
some text that shouldn't be indented
*fake heading
- bullet point 1
- bullet point 2
- bullet point 3
- **bold bullet point**
    - sub bullet point 1
        - sub sub bullet point 1
        - sub sub bullet point 2
            - sub sub sub bullet point 1
   -fake bullet point

1. order list entry 1
2. order list entry 2
some text in between with bold italic ***highlights***.

1. level 1 order list entry 1
2. level 1 order list entry 2
    1. level 2 order list entry 1

1. order list entry 1
- unorder list entry 1 with a [link](link.md)
1. order list entry 1

1. level 1 order list entry 1
2. level 1 order list entry 2
    1. level 2 order list entry 1
## To Do 
normal text.
[a link to a file](a link to a file.md)
*italic* and **bold** text!
## Checklist
- [ ] unchecked
- [x] checked
- [ ] pending (on hold)
- [ ] more info needed (needs further input/clarification)
- [ ] dropped (put down/cancelled)
- [ ] not done (didn't do)
```
const var = 'this is some code';
console.log(var);
```
``` rust
let var = "this is some better code";
println!("{var}");
```
